gadget/main.c

```diff
@@ -98,23 +100,23 @@ int main(int argc, char **argv)
     }
 
     if(RestartFlag == 1) {
-        RestartSnapNum = find_last_snapnum();
-        message(0, "Last Snapshot number is %d.\n", RestartSnapNum);
+        /* Last snapshot will be detected in begrun*/
+        RestartSnapNum = -2;
     }
 
     /*Set up GSL so it gives a proper MPI termination*/
     gsl_set_error_handler(gsl_handler);
 
     /*Initialize the memory manager*/
-    mymalloc_init(All.MaxMemSizePerNode);
+    mymalloc_init(MaxMemSizePerNode);
 
     /* Make sure memory has finished initialising on all ranks before doing more.
      * This may improve stability */
     MPI_Barrier(MPI_COMM_WORLD);
 
-    init_endrun(All.ShowBacktrace);
+    init_endrun(ShowBacktrace);
 
-    begrun(RestartSnapNum);
+    RestartSnapNum = begrun(RestartSnapNum);
```

Comment:
What about passing RestartSnapFlag annd RestartSnapNum both to begrun to avoid defining a new magic number -2?

Reply:
Yes, good idea.